dynomite/src/ext.rs

```diff
@@ -0,0 +1,197 @@
+//!Eextention interfaces for rusoto `DynamoDb`
+
+use std::collections::HashMap;
+
+use futures::{stream, Future, Stream};
+use rusoto_dynamodb::{
+  AttributeValue, BackupSummary, DynamoDb, ListBackupsError, ListBackupsInput, ListTablesError,
+  ListTablesInput, QueryError, QueryInput, ScanError, ScanInput,
+};
+
+/// Exention methods for DynamoDb implementations
+pub trait DynamoDbExt {
+  // see https://github.com/boto/botocore/blob/5250e2e7a3209eb995283ac018aea37d3bc1da45/botocore/data/dynamodb/2012-08-10/paginators-1.json
+
+  /// A `Stream` oriented version of `list_backups`
+  fn stream_list_backups(
+    &'static self,
```

Comment:
Some other options https://stackoverflow.com/questions/48613350/how-to-return-a-future-combinator-with-self